spec/unit/pops/types/type_calculator_spec.rb

```diff
@@ -832,10 +885,40 @@ class Foo
         tested_types = all_types - [
           Puppet::Pops::Types::PAnyType,
           Puppet::Pops::Types::POptionalType,
+          Puppet::Pops::Types::PNotUndefType,
           Puppet::Pops::Types::PDataType] - collection_types
         t = Puppet::Pops::Types::PStructType.new()
         tested_types.each {|t2| t.should_not be_assignable_to(t2.new) }
       end
+
+      it 'Default key optionality is controlled by value assignability to undef' do
```

Comment:
This example title does not seem to describe what the test does as it checks if the constructed type has the expected shape.